src/LLVM.jl

```diff
@@ -5,10 +5,63 @@ using Printf
 using Libdl
 
 
-const ext = joinpath(@__DIR__, "..", "deps", "ext.jl")
-isfile(ext) || error("LLVM.jl has not been built, please run Pkg.build(\"LLVM\").")
-include(ext)
-const libllvm = libllvm_path
+## discovery
+
+using Libdl
+
+VERSION >= v"0.7.0-DEV.2576" || error("This version of LLVM.jl requires Julia 0.7")
+
+# figure out the path to libLLVM by looking at the libraries loaded by Julia
+const libllvm_paths = filter(Libdl.dllist()) do lib
+    occursin("LLVM", basename(lib))
+end
+if isempty(libllvm_paths)
+    error("""
+        Cannot find the LLVM library loaded by Julia.
+        Please use a version of Julia that has been built with USE_LLVM_SHLIB=1 (like the official binaries).
+        If you are, please file an issue and attach the output of `Libdl.dllist()`.""")
+end
+if length(libllvm_paths) > 1
+    error("""
+        Multiple LLVM libraries loaded by Julia.
+        Please file an issue and attach the output of `Libdl.dllist()`.""")
+end
+const libllvm = first(libllvm_paths)
```

Comment:
Maybe it makes sense to mark this `Base.include_dependency` so that when the lib changes we recompile?

Reply:
Huh, I did not know about that mechanism! Very useful, especially for the CUDA stuff. I'm not sure it is needed here, since an upgrade of LLVM (implying Julia changed) should trigger recompilation too, but it can't hurt.

Reply:
Haha, I only remember it because I had spend to much time in `base/loading.jl`.